### PR TITLE
Add creation of database using http instead of client library

### LINKED ISSUE #117: Create database if it doesn't exist.
It would be useful to have this enhancement in InfluxDB publisher. Database could be created if doesn't already exist.

### PR DESCRIPTION
**Pull request by mkuculyma**
Fixes #117 

Summary of changes:
- Added creation of database using http package instead of inluxdb client lib to workaround https://github.com/influxdata/influxdb/issues/8108 

How to verify it:
- run tests

Testing done:
- small

### CODE REVIEW

File: influxdb/influxdb.go [hunk 1]
```diff
@@ -371,13 +373,66 @@ type clientConnection struct {
...
 // Create database if it doesn't exist
-func (c *clientConnection) initDB(db string) error {
-	q := client.Query{
-		Command:  fmt.Sprintf("CREATE DATABASE %s", db),
-		Database: db,
+// workaround: use http instead of client library because of the issue
+// ref: https://github.com/influxdata/influxdb/issues/8108
+func (c *clientConnection) initDB(u *url.URL, user, pass, db string) error {
+
+	//urlStr := fmt.Sprintf("%s/query?q=CREATE+DATABASE+%s", u.String(), db)
```
Comment (PatrykMatyjasek) [inline influxdb/influxdb.go]: please remove commented code and empty line

File: influxdb/influxdb.go [hunk 2]
```diff
@@ -371,13 +373,64 @@ type clientConnection struct {
...
+		return err
+	}
+
+	return nil
+}
+
+// Check if database exists
+// workaround: use http instead of client library because of the issue
+// ref: https://github.com/influxdata/influxdb/issues/8108
+func (c *clientConnection) dbExists(u *url.URL, user, pass, db string) bool {
```
Comment (andrzej-k) [inline influxdb/influxdb.go]: Please remember to closer resp.Body at the end.

Comment (mkuculyma) [inline influxdb/influxdb.go]: Fixed

File: influxdb/influxdb.go [hunk 3]
```diff
@@ -371,13 +373,68 @@ type clientConnection struct {
...
+	req.URL.RawQuery = params.Encode()
+
+	req.SetBasicAuth(user, pass)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return false
+	}
+
+	bodyText, err := ioutil.ReadAll(resp.Body)
```
Comment (andrzej-k) [inline influxdb/influxdb.go]: What happens when resp.Body is nil?

**Pull request merged**